backend/tarsy/integrations/mcp/client.py

```
@@ -163,16 +164,40 @@ async def list_tools(
                             all_tools[server_name] = tools_result.tools
                             
                             # Log the successful response
                             self._log_mcp_list_tools_response(server_name, tools_result.tools, request_id)
                             break  # Success, exit retry loop
+                        
+                        except asyncio.TimeoutError:
+                            error_msg = f"List tools timed out after {timeout_seconds}s (attempt {attempt + 1}/{max_retries})"
+                            logger.error(f"{error_msg} for {server_name}")
+                            
+                            if attempt < max_retries - 1:
+                                logger.warning(f"Retrying list_tools after timeout...")
```

Ruff (0.14.0)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)